packages/scandipwa/src/route/MenuPage/MenuPage.container.js

```diff
@@ -34,43 +33,42 @@ export const mapDispatchToProps = (dispatch) => ({
 });
 
 /** @namespace Route/MenuPage/Container */
+
 export class MenuPageContainer extends PureComponent {
-    static propTypes = {
-        updateMeta: PropTypes.func.isRequired,
-        history: HistoryType.isRequired,
-        changeHeaderState: PropTypes.func.isRequired,
-        isMobile: PropTypes.bool.isRequired
-    };
+     static propTypes = {
+         updateMeta: PropTypes.func.isRequired,
+         changeHeaderState: PropTypes.func.isRequired,
+         isMobile: PropTypes.bool.isRequired
+     };
+
+     componentDidMount() {
+         const {
+             updateMeta,
+             changeHeaderState,
+             isMobile
+         } = this.props;
 
-    componentDidMount() {
-        const { updateMeta, changeHeaderState } = this.props;
-        updateMeta({ title: __('Menu') });
-        this.redirectIfNotOnMobile();
-        changeHeaderState({
-            name: MENU,
-            onBackClick: () => history.goBack()
-        });
-    }
+         if (isMobile) {
+             updateMeta({ title: __('Menu') });
```

Comment:
Can you please make it more clear and less nesting, so if !isMobile then redirect and if is mobile, then do all things for page.